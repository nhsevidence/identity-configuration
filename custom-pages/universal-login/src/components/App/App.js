import React from 'react'
import Header from "../Header/Header";
import {Route, Switch} from "react-router";
import {BrowserRouter as Router} from "react-router-dom";
import Login from "../Login/Login";
import ForgotPassword from "../ForgotPassword/ForgotPassword";
import ForgotPasswordSuccess from "../ForgotPasswordSuccess/ForgotPasswordSuccess";
import Register from "../Register/Register";
import RegisterSuccess from "../RegisterSuccess/RegisterSuccess";
import ResetPassword from "../../components/ResetPassword/ResetPassword";
import ResetPasswordSuccess from "../../components/ResetPasswordSuccess/ResetPasswordSuccess";
import NotFound from "../NotFound/NotFound";
import './App.scss'

class App extends React.Component {
  render() {
    return (
      <div>
        <div className="wrapper">
          <Header className="col leftCol" />
          <div className="mainContainer col rightCol">
            <Router>
                <Switch>
                  {process.env.REACT_APP_RENDER === 'login' ?
                    <Route exact path="/" component={Login} /> :
                    <Route exact path="/" component={ResetPassword} />
                  }
                  <Route path='/login' component={Login} />
                  <Route path='/register' component={Register} />
                  <Route path='/regsuccess' component={RegisterSuccess} />
                  <Route path='/forgotPassword' component={ForgotPassword} />
                  <Route path='/forgotsuccess' component={ForgotPasswordSuccess} />
                  <Route path='/resetpassword' component={ResetPassword} />
                  <Route path='/change-password' component={ResetPassword} />
                  <Route path='/resetsuccess' component={ResetPasswordSuccess} />
                  <Route component={NotFound} />
                </Switch>
            </Router>
          </div>
        </div>
      </div>
    )
  }
}
export default App
