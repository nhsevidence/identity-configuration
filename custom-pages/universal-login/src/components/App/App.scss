@import "~@nice-digital/nds-forms/scss/forms";
@import "~@nice-digital/nds-button/scss/button";

:root {
  --font-base: Lato, Helvetica, sans-serif;
  --gutter: 1.2rem;
  --col-width-mobile: 20rem;
  --col-width-desktop: 36rem;
  --top: 3.6rem;
}

html,
body {
  margin: 0;
  padding: 0;
  font-family: var(--font-base);
  background-color: #fff;
  font-size: 16px;
}

.wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.col {
  height: 100%;
  padding: 3rem;
  width: var(--col-width-desktop);
}

.rightCol {
  padding-top: 1.4rem !important;
}

.mainContainer {
  padding-top: var(--top);
  min-width: var(--col-width-desktop);
}

@media (max-width: 700px) {
  .wrapper {
    flex-direction: column;
  }
  .rightCol,
  .leftCol {
    width: 100%;
    min-width: var(--col-width-mobile);
    padding-bottom: 0;
  }
  .mainContainer {
    min-width: var(--col-width-mobile);
  }

  .navigation {
    width: calc(var(--col-width-mobile) - 5rem);
  }
}

.btn {
  text-align: center !important;
  margin: 0 !important;
}

legend{
  border: none !important;
  width: auto !important;
}

label{
  color: #333 !important;
}
