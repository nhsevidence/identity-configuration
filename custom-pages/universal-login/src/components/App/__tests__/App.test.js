import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import App from '../App'

Enzyme.configure({ adapter: new Adapter() });

describe('App components', () => {

  beforeEach(() => {
    jest.resetModules()
  });

  it('should render Login components correctly', () => {
    process.env.REACT_APP_RENDER = 'login';
    expect(shallow(<App />)).toMatchSnapshot()
  });

  it('should render ResetPassword components correctly', () => {
    process.env.REACT_APP_RENDER = 'resetpassword';
    expect(shallow(<App />)).toMatchSnapshot()
  })
});
