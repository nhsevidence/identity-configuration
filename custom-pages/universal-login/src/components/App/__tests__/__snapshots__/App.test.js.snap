// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components should render Login components correctly 1`] = `ShallowWrapper {}`;

exports[`App components should render ResetPassword components correctly 1`] = `ShallowWrapper {}`;
